### ISSUE TITLE
Code from `tutorials\examples\ptl-trainer\litgpt_ptl_small.py` not working, model not learning

### ISSUE THREAD

**Issue by li-yang23** [Lightning-AI/litgpt#2151]
`litgpt==0.5.11 litdata==0.2.58, OS: windows 10, python==3.13, torch==2.9.0+cu126`
I'm following the tutorials to pretrain a smollm2-135M with tinystories dataset, and my script is like this:

```python
import lightning as L
import torch

from litgpt import LLM, Tokenizer
from litgpt.data import TinyStories
class LitLLM(L.LightningModule):
    def __init__(self, checkpoint_dir, tokenizer_dir=None, trainer_ckpt_path=None):
        super().__init__()

        self.llm = LLM.load(checkpoint_dir, tokenizer_dir=tokenizer_dir, distribute=None)
        self.trainer_ckpt_path = trainer_ckpt_path

    def setup(self, stage):
        self.llm.trainer_setup(trainer_ckpt=self.trainer_ckpt_path)

    def training_step(self, batch):
        ############
        # this code is to see the change of model parameter.
        ############
        data = self.llm.model.lm_head.weight
        mean   = data.mean().item()
        std    = data.std().item()
        minv   = data.min().item()
        maxv   = data.max().item()
        rms    = data.square().mean().sqrt().item()
        nan    = torch.isnan(data).sum().item()
        inf    = torch.isinf(data).sum().item()

        msg = (f"lr={self.lr_schedulers().get_last_lr()} "
               f"mean={mean:+.6f} std={std:.6f} "
               f"min={minv:+.6f} max={maxv:+.6f} rms={rms:.6f} "
               f"nan={nan} inf={inf}")
        print(msg)
        ############
        # code above is to see the change of model parameter.
        ############
        logits, loss = self.llm(input_ids=batch[:, :-1].contiguous().long(), target_ids=batch[:, 1:].contiguous().long()) # I change batch['input_ids'] to batch[:, : -1] since tinystories does not have 'input_ids' and 'labels' keys after tokenization
        self.log("train_loss", loss, prog_bar=True)
        return loss

    def validation_step(self, batch):
        # print(batch)
        logits, loss = self.llm(input_ids=batch[:, :-1].contiguous().long(), target_ids=batch[:, 1:].contiguous().long())
        self.log("validation_loss", loss, prog_bar=True)
        return loss

    def configure_optimizers(self):
        warmup_steps = 10
        optimizer = torch.optim.AdamW(self.llm.model.parameters(), lr=0.0002, weight_decay=0.0, betas=(0.9, 0.95))
        scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lambda step: step / warmup_steps)
        return [optimizer], [scheduler]
lit_model = LitLLM(checkpoint_dir="smlolm2-135m-random-weights", tokenizer_dir="HuggingFaceTB/SmolLM2-135M")
lit_model = LitLLM(checkpoint_dir="smlolm2-135m-random-weights", tokenizer_dir="HuggingFaceTB/SmolLM2-135M")
data = TinyStories()
data.connect(lit_model.llm.tokenizer, batch_size=1, max_seq_length=512)
batch_size = 8
accumulate_grad_batches = 1
trainer = L.Trainer(
    devices=1,
    accelerator="cuda",
    max_epochs=1,
    accumulate_grad_batches=accumulate_grad_batches,
    precision="bf16-true",
)
trainer.fit(lit_model, data)
```

and the output is like this

```shell

Epoch 0:   0%|          | 0/1793507 [00:00<?, ?it/s]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 1/1793507 [00:02<1175:20:35,  0.42it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 2/1793507 [00:02<611:40:05,  0.81it/s, v_num=12, train_loss=10.90] lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 3/1793507 [00:02<420:23:11,  1.19it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 4/1793507 [00:02<325:11:14,  1.53it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 5/1793507 [00:02<267:31:46,  1.86it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 6/1793507 [00:02<229:05:09,  2.17it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 7/1793507 [00:02<201:38:04,  2.47it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 8/1793507 [00:02<181:02:59,  2.75it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 9/1793507 [00:02<165:04:52,  3.02it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 10/1793507 [00:03<152:14:21,  3.27it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 11/1793507 [00:03<141:53:25,  3.51it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 12/1793507 [00:03<133:09:19,  3.74it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 13/1793507 [00:03<125:55:43,  3.96it/s, v_num=12, train_loss=11.00]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 14/1793507 [00:03<119:49:29,  4.16it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 15/1793507 [00:03<114:52:02,  4.34it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 16/1793507 [00:03<110:05:26,  4.53it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 17/1793507 [00:03<105:53:13,  4.70it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 18/1793507 [00:03<102:08:18,  4.88it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 19/1793507 [00:03<98:58:22,  5.03it/s, v_num=12, train_loss=10.80] lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 20/1793507 [00:03<95:54:51,  5.19it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 21/1793507 [00:03<93:08:15,  5.35it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 22/1793507 [00:04<90:36:49,  5.50it/s, v_num=12, train_loss=10.80]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 23/1793507 [00:04<88:19:32,  5.64it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 24/1793507 [00:04<86:16:08,  5.77it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
...
Epoch 0:   0%|          | 121/1793507 [00:12<49:50:26, 10.00it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 122/1793507 [00:12<49:44:43, 10.01it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 123/1793507 [00:12<49:39:34, 10.03it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 124/1793507 [00:12<49:34:14, 10.05it/s, v_num=12, train_loss=10.90]lr=[0.0] mean=-0.000124 std=0.024048 min=-0.041748 max=+0.041504 rms=0.024048 nan=0 inf=0
Epoch 0:   0%|          | 125/1793507 [00:12<49:35:06, 10.05it/s, v_num=12, train_loss=10.90]
```

as shown, learning rate is 0.0 all the time, and the parameter is not changing, the model is not running. Is there any other guys meet this problem?

**Comment by li-yang23** [comment]
I found the potential problem is the `configure_optimizers` function, I changed

```python
optimizer = torch.optim.AdamW(self.llm.model.parameters(), lr=0.0002, weight_decay=0.0, betas=(0.9, 0.95))
scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lambda step: step / warmup_steps)
```
to
```python
optimizer = torch.optim.AdamW(self.parameters(), lr=0.0002, weight_decay=0.0, betas=(0.9, 0.95))
scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lambda step: (step+1) / warmup_steps)
```
and it (seems) works, the training loss and the mean or std of weights start to change.

**Comment by bhimrazy** [comment]
Closing this since it’s resolved — please feel free to reopen if the issue persists.
